tests/optim/param/test_images.py

```diff
@@ -242,5 +247,316 @@ def test_laplacianimage_random_forward(self) -> None:
+        image_param = images.SharedImage(
+            shapes=shared_shapes, parameterization=test_param
+        )
+
+        offset_vals = ((1, 2, 3, 4), (4, 3, 2, 1), (1, 2, 3, 4))
+        offset = image_param.get_offset(offset_vals, 3)
+
+        self.assertEqual(len(offset), 3)
+        self.assertEqual(offset, [[int(o) for o in v] for v in offset_vals])
+
+    def test_sharedimage_get_offset_single_set_four_numbers(self) -> None:
+        if torch.__version__ == "1.2.0":
+            raise unittest.SkipTest(
+                "Skipping SharedImage test due to insufficient Torch version."
+            )
+        shared_shapes = (128 // 2, 128 // 2)
+        test_param = lambda: torch.ones(3, 3, 224, 224)  # noqa: E731
+        image_param = images.SharedImage(
+            shapes=shared_shapes, parameterization=test_param
+        )
+
+        offset_vals = (1, 2, 3, 4)
+        offset = image_param.get_offset(offset_vals, 3)
+
+        self.assertEqual(len(offset), 3)
+        self.assertEqual(offset, [list(offset_vals)] * 3)
+
+    def test_sharedimage_get_offset_single_set_three_numbers(self) -> None:
+        if torch.__version__ == "1.2.0":
+            raise unittest.SkipTest(
+                "Skipping SharedImage test due to insufficient Torch version."
+            )
+        shared_shapes = (128 // 2, 128 // 2)
+        test_param = lambda: torch.ones(3, 3, 224, 224)  # noqa: E731
+        image_param = images.SharedImage(
+            shapes=shared_shapes, parameterization=test_param
+        )
+
+        offset_vals = (2, 3, 4)
+        offset = image_param.get_offset(offset_vals, 3)
+
+        self.assertEqual(len(offset), 3)
+        self.assertEqual(offset, [[0] + list(offset_vals)] * 3)
+
+    def test_sharedimage_get_offset_single_set_two_numbers(self) -> None:
+        if torch.__version__ == "1.2.0":
+            raise unittest.SkipTest(
+                "Skipping SharedImage test due to insufficient Torch version."
+            )
+        shared_shapes = (128 // 2, 128 // 2)
+        test_param = lambda: torch.ones(3, 3, 224, 224)  # noqa: E731
+        image_param = images.SharedImage(
+            shapes=shared_shapes, parameterization=test_param
+        )
+
+        offset_vals = (3, 4)
+        offset = image_param.get_offset(offset_vals, 3)
+
+        self.assertEqual(len(offset), 3)
+        self.assertEqual(offset, [[0, 0] + list(offset_vals)] * 3)
+
+    def apply_offset_compare(
+        self, x_list: List[torch.Tensor], offset_list: List[List[int]]
+    ) -> List[torch.Tensor]:
+        A = []
+        for x, offset in zip(x_list, offset_list):
+            assert x.dim() == 4
+            size = list(x.size())
+
+            offset_pad = (
+                [[abs(offset[0])] * 2]
+                + [[abs(offset[1])] * 2]
+                + [[abs(offset[2])] * 2]
+                + [[abs(offset[3])] * 2]
+            )
+
+            x = SymmetricPadding.apply(x, offset_pad)

```

Comment:
I think the test cases perform forward pass only. We'd need to call backward or autograd to test backward pass as well, don't we ?

Reply:
I've added a test for the backwards pass!

Reply:
Thank you!